BEAMS3D/Sources/beams3d_physics_mod.f90

```diff
@@ -305,8 +307,8 @@ SUBROUTINE beams3d_physics_gc(t, q)
             zeff_temp = max(fval(1),one)
             CALL R8HERM3FCN(ict,1,1,fval,i,j,k,xparam,yparam,zparam,&
                             hr(i),hri(i),hp(j),hpi(j),hz(k),hzi(k),&
-                            S4D(1,1,1,1),nr,nphi,nz)
-            s_temp = max(fval(1),zero)
+                            RHO4D(1,1,1,1),nr,nphi,nz)
+            rho_temp = max(fval(1),zero)
             DO l = 1, NION
```

Comment:
This should be improved further. It's generally not good enough to do a single interpolation via RHO4D. It's much better to get XRHO4D and YRHO4D, then take the sqrt to get rho.

The RHO4D function might not even have an 'axis' (loci of points where rho=0), but sqrt(XRHO4D^2 + YRHO4D^2) will since they boundary surfaces between positive and negative values.

Reply:
Same with other single S4D or RHO4D lookups in this and similar files.